projects/CNSPRCY/services/cnsprcy/module.nix

```
@@ -0,0 +1,73 @@
+{
+  config,
+  lib,
+  pkgs,
+  ...
+}:
+let
+  cfg = config.services.cnsprcy;
+  stateDir = "/var/lib/cnsprcy";
+in
+{
+  options.services.cnsprcy = {
+    enable = lib.mkEnableOption "cnsprcy";
+    package = lib.mkPackageOption pkgs "cnsprcy" { };
+
+    hostname = lib.mkOption {
+      type = lib.types.str;
+      description = "Hostname of CNSPRCY server";
+      default = builtins.getEnv "HOSTNAME";
+    };
+
+    user = lib.mkOption {
+      type = lib.types.str;
+      description = "Username of the system user that should own files and services related to CNSPRCY.";
+      default = "cnsprcy";
+    };
+    group = lib.mkOption {
+      type = lib.types.str;
+      description = "Group that contains the system user that executes CNSPRCY.";
+      default = "cnsprcy";
+    };
+  };
+
+  config = lib.mkIf cfg.enable {
+
+    environment.systemPackages = [ cfg.package ];
+
+    users.users."${cfg.user}" = {
+      isSystemUser = true;
+      group = cfg.group;
+
+      # CNSPRCY looks for the user home to initialize its data directory
+      # at $HOME/.local/share/cnsprcy, but systemd.tmpfiles needs to create
+      # the dir otherwise permissions won't be set correctly.
+      home = "${stateDir}";
+      useDefaultShell = true;
+    };
+    users.groups."${cfg.group}" = { };
+
+    systemd.tmpfiles.rules = [
+      "d ${stateDir}/.local/share/cnsprcy 0700 ${cfg.user} ${cfg.group} -"
+      "d ${stateDir}/.local/share/cnsprcy/handlers 0700 ${cfg.user} ${cfg.group} -"
+    ];
+
+    systemd.services.cnsprcy = {
+      description = "CNSPRCY service";
+      wantedBy = [ "multi-user.target" ];
+      after = [ "nss-user-lookup.target" ];
```

The service should probably only start after the tmpfiles have been created:

```suggestion
      after = [
        "nss-user-lookup.target"
        "systemd-tmpfiles-setup.service"
      ];
```